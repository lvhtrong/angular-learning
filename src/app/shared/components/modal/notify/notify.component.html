<div *ngIf="title" class="modal-header">
  <h4 class="modal-title">{{ title }}</h4>
</div>
<div class="modal-body">
  <p>{{ content }}</p>
</div>
<div class="modal-footer">
  <button
    type="button"
    class="btn btn-outline-dark"
    (click)="activeModal.close('Close click')"
  >
    {{ buttonText }}
  </button>
</div>
