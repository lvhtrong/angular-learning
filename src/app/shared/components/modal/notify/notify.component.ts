import { Component, OnInit, Input } from '@angular/core';
import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';

@Component({
  selector: 'app-modal-notify',
  templateUrl: './notify.component.html',
  styleUrls: ['./notify.component.scss'],
})
export class ModalNotifyComponent implements OnInit {
  @Input() title = '';
  @Input() content = '';
  @Input() buttonText = 'Close';

  constructor(public activeModal: NgbActiveModal) {}

  ngOnInit(): void {}
}
