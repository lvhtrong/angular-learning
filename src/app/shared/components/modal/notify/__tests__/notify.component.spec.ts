import {
  Spectator,
  createComponentFactory,
  byText,
} from '@ngneat/spectator/jest';
import { MockService } from 'ng-mocks';
import { ModalNotifyComponent } from '../notify.component';
import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';

const createComponent = createComponentFactory({
  component: ModalNotifyComponent,
  providers: [
    {
      provide: NgbActiveModal,
      useValue: MockService(NgbActiveModal),
    },
  ],
});

const setup = (): Spectator<ModalNotifyComponent> => {
  return createComponent();
};

it('should render as default', () => {
  const spectator = setup();
  spectator.setInput('title', 'sample title');
  spectator.setInput('content', 'sample content');
  spectator.setInput('buttonText', 'sample button text');

  expect(spectator.element).toMatchSnapshot();
});

describe('tap Close button', () => {
  it('should trigger to hide modal', () => {
    const spectator = setup();

    const closeButton = spectator.query(byText('Close'));
    spectator.click(closeButton);

    const activeModal = spectator.inject(NgbActiveModal);
    expect(activeModal.close).toHaveBeenCalledWith('Close click');
  });
});
