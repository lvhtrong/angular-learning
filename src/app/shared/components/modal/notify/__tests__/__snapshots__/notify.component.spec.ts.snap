// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render as default 1`] = `
<div
  id="root0"
  ng-version="9.1.7"
>
  
  <div
    class="modal-header"
  >
    <h4
      class="modal-title"
    >
      sample title
    </h4>
  </div>
  <div
    class="modal-body"
  >
    <p>
      sample content
    </p>
  </div>
  <div
    class="modal-footer"
  >
    <button
      class="btn btn-outline-dark"
      type="button"
    >
       sample button text 
    </button>
  </div>
</div>
`;
